### ISSUE TITLE
Error when mirroring data to BigQuery: Resource did not meet condition IF_MATCH

### ISSUE THREAD

**Issue by guilhermeagostinelli** [firebase/extensions#606]
- Extension name: `firestore-bigquery-export`
- Extension version: v0.1.12

It was running perfectly fine on v0.1.5 but after upgrading to v0.1.12 the error below has already appeared twice in the logs. There have been executions before and after the error which executed successfully.

```
Error when mirroring data to BigQuery { Error: Resource positiv-app:firestore_export_conclusions.conclusions_raw_changelog did not meet condition IF_MATCH
    at new ApiError (/workspace/node_modules/@google-cloud/common/build/src/util.js:58:15)
    at Util.parseHttpRespBody (/workspace/node_modules/@google-cloud/common/build/src/util.js:193:38)
    at Util.handleResp (/workspace/node_modules/@google-cloud/common/build/src/util.js:134:117)
    at retryRequest (/workspace/node_modules/@google-cloud/common/build/src/util.js:432:22)
    at onResponse (/workspace/node_modules/retry-request/index.js:214:7)
    at /workspace/node_modules/teeny-request/build/src/index.js:233:13
    at process._tickCallback (internal/process/next_tick.js:68:7)
  code: 412,
 ```

**Comment by dackers86** [comment]
Hi @guilhermeagostinelli 

A history of this issue can be found here https://github.com/firebase/extensions/issues/235

The original fix is here https://github.com/firebase/extensions/pull/398. 

Can you confirm that you have a `document_id ` column in your BQ tables?

**Comment by dackers86** [comment]
Also could you confirm this is a recurring issue?

There seems to be a known issue when first implemented https://github.com/firebase/extensions/pull/398#issuecomment-666447313

**Comment by guilhermeagostinelli** [comment]
Hi @dackers86 , thanks for the reply.

> Can you confirm that you have a document_id column in your BQ tables?

Yes, the column is there and, by the way, all documents before the migration date have a NULL value.

> Also could you confirm this is a recurring issue?

It was first seen on 2/19/21 and last seen on 2/21/21
There have been 3 occurrences until now.

**Comment by dackers86** [comment]
Thanks @guilhermeagostinelli.

Although not ideal, could clearing your BQ tables and re-syncing solves this particular case?

Additionally, do you have any errors in the extensions logs? Hopefully they could provide a hint on any issues that are occurring?

**Comment by SuryaSrikanthsml** [comment]
Here is the error in extension's cloud function logs

Error when mirroring data to BigQuery ApiError: Precondition check failed. at new ApiError (/workspace/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/node_modules/@google-cloud/common/build/src/util.js:58:15) at Util.parseHttpRespBody (/workspace/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/node_modules/@google-cloud/common/build/src/util.js:193:38) at Util.handleResp (/workspace/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/node_modules/@google-cloud/common/build/src/util.js:134:117) at /workspace/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/node_modules/@google-cloud/common/build/src/util.js:432:22 at onResponse (/workspace/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/node_modules/retry-request/index.js:228:7) at /workspace/node_modules/@firebaseextensions/firestore-bigquery-change-tracker/node_modules/teeny-request/build/src/index.js:233:13 at processTicksAndRejections (internal/process/task_queues.js:95:5) { code: 412, errors: [ { message: 'Precondition check failed.', domain: 'global', reason: 'failedPrecondition' } ], response: Gunzip { _writeState: Uint32Array(2) [ 16110, 0 ], _readableState: ReadableState { objectMode: false, highWaterMark: 16384, buffer: BufferList { head: null, tail: null, length: 0 }, length: 0, pipes: [], flowing: true, ended: true, endEmitted: true, reading: false, sync: false, needReadable: false, emittedReadable: false, readableListening: false, resumeScheduled: false, errorEmitted: false, emitClose: true, autoDestroy: true, destroyed: true, errored: null, closed: true, closeEmitted: true, defaultEncoding: 'utf8', awaitDrainWriters: null, multiAwaitDrain: false, readingMore: false, dataEmitted: true, decoder: null, encoding: null, [Symbol(kPaused)]: false }, _events: [Object: null prototype] { prefinish: [Function: prefinish], error: [Array], data: [Function (anonymous)], end: [Function (anonymous)] }, _eventsCount: 4, _maxListeners: undefined, _writableState: WritableState { objectMode: false, highWaterMark: 16384, finalCalled: false, needDrain: false, ending: true, ended: true, finished: true, destroyed: true, decodeStrings: true, defaultEncoding: 'utf8', length: 0, writing: false, corked: 0, sync: false, bufferProcessing: false, onwrite: [Function: bound onwrite], writecb: null, writelen: 0, afterWriteTickInfo: null, buffered: [], bufferedIndex: 0, allBuffers: true, allNoop: true, pendingcb: 0, prefinished: true, errorEmitted: false, emitClose: true, autoDestroy: true, errored: null, closed: true }, allowHalfOpen: true, bytesWritten: 171, _handle: null, _outBuffer: <Buffer 7b 0a 20 20 22 65 72 72 6f 72 22 3a 20 7b 0a 20 20 20 20 22 63 6f 64 65 22 3a 20 34 31 32 2c 0a 20 20 20 20 22 6d 65 73 73 61 67 65 22 3a 20 22 50 72 ... 16334 more bytes>, _outOffset: 274, _chunkSize: 16384, _defaultFlushFlag: 2, _finishFlushFlag: 2, _defaultFullFlushFlag: 3, _info: undefined, _maxOutputLength: 4294967295, _level: -1, _strategy: 0, statusCode: 412, statusMessage: 'Precondition Failed', request: { agent: [Agent], headers: [Object], href: 'https://bigquery.googleapis.com/bigquery/v2/projects/<PROJECT NAME>/datasets/firestore_export/tables/user_programs_raw_latest?' }, body: '{\n' + ' "error": {\n' + ' "code": 412,\n' + ' "message": "Precondition check failed.",\n' + ' "errors": [\n' + ' {\n' + ' "message": "Precondition check failed.",\n' + ' "domain": "global",\n' + ' "reason": "failedPrecondition"\n' + ' }\n' + ' ],\n' + ' "status": "FAILED_PRECONDITION"\n' + ' }\n' + '}\n', headers: { 'alt-svc': 'h3=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"', 'cache-control': 'private', 'content-encoding': 'gzip', 'content-type': 'application/json; charset=UTF-8', date: 'Wed, 16 Mar 2022 08:10:58 GMT', etag: '5DrnI3mBip62BTyhHue98w==', server: 'ESF', 'transfer-encoding': 'chunked', vary: 'Origin, X-Origin, Referer', 'x-content-type-options': 'nosniff', 'x-frame-options': 'SAMEORIGIN', 'x-xss-protection': '0' }, toJSON: [Function: toJSON], [Symbol(kCapture)]: false, [Symbol(kTransformState)]: { afterTransform: [Function: bound afterTransform], needTransform: false, transforming: false, writecb: null, writechunk: null, writeencoding: 'buffer' }, [Symbol(kError)]: null } }

**Comment by oliverkaisti** [comment]
same Error as well here as  @SuryaSrikanthsml

**Comment by SuryaSrikanthsml** [comment]
I have downgraded to 0.1.20, it is working without any errors

**Comment by dgilperez** [comment]
I am seeing this as well

**Comment by famaranon** [comment]
> Same problem as @SuryaSrikanthsml. Downgraded to 0.1.20 worked.

@dperezrada how did you downgrade the extension ? Did you manually install it ?

**Comment by dperezrada** [comment]
@famaranon you can define the extension version in the `firebase.json` take a look to the following link: https://firebase.google.com/docs/extensions/reuse-project-config

**Comment by madmacc** [comment]
I had this issue and it seems to be resolved after upgrading from 0.1.24 to 0.1.30

**Comment by madmacc** [comment]
I take that back it has just started occurring again for a different collection/table.
